File: app/schemas/episode.py
from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class EpisodeBBox(BaseModel):
    minx: float
    miny: float
    maxx: float
    maxy: float


class EpisodeSlideItem(BaseModel):
    type: str
    thumbnail_url: Optional[str] = None
    url: Optional[str] = None
    satellite_image_id: Optional[str] = None
    generated_at: Optional[str] = None


class FireEpisodeListItem(BaseModel):
    id: UUID
    status: str
    start_date: datetime
    end_date: Optional[datetime] = None
    last_seen_at: Optional[datetime] = None
    bbox: Optional[EpisodeBBox] = None
    centroid_lat: Optional[float] = None
    centroid_lon: Optional[float] = None
    provinces: Optional[List[str]] = None
    event_count: int
    detection_count: int
    frp_sum: Optional[float] = None
    frp_max: Optional[float] = None
    estimated_area_hectares: Optional[float] = None
    gee_candidate: bool = False
    gee_priority: Optional[int] = None
    slides_data: Optional[List[EpisodeSlideItem]] = None
    representative_event_id: Optional[UUID] = None

    model_config = ConfigDict(from_attributes=True)


class FireEpisodeEventItem(BaseModel):
    id: UUID
    start_date: datetime
    end_date: datetime
    province: Optional[str] = None
    total_detections: Optional[int] = None
    max_frp: Optional[float] = None
    estimated_area_hectares: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)


class FireEpisodeDetail(FireEpisodeListItem):
    events: List[FireEpisodeEventItem] = []


class FireEpisodeListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    total_pages: int
    episodes: List[FireEpisodeListItem]

File: frontend/src/types/episode.ts
export type EpisodeStatus = 'active' | 'monitoring' | 'extinct' | 'closed'

export type EpisodeSlide = {
  type: string
  thumbnail_url?: string | null
  url?: string | null
  satellite_image_id?: string | null
  generated_at?: string | null
}

export type EpisodeListItem = {
  id: string
  status?: EpisodeStatus
  start_date: string
  end_date?: string | null
  last_seen_at?: string | null
  centroid_lat?: number | null
  centroid_lon?: number | null
  provinces?: string[] | null
  event_count: number
  detection_count: number
  frp_sum?: number | null
  frp_max?: number | null
  estimated_area_hectares?: number | null
  gee_candidate?: boolean
  gee_priority?: number | null
  slides_data?: EpisodeSlide[] | null
  /** pending | processing | ready | failed */
  slides_status?: string | null
  representative_event_id?: string | null
  is_recent?: boolean
  recent_days?: number | null
  recovery_status?: string | null
  is_potential_violation?: boolean
}

export type EpisodeListResponse = {
  total: number
  page: number
  page_size: number
  total_pages: number
  episodes: EpisodeListItem[]
}
